### ISSUE TITLE
Customize Users

### ISSUE THREAD

**Issue by lfontaine974** [PiranhaCMS/piranha.core#822]
Hi,

Is it possible to add custom data to users, for example to add others general informations ? Same with extensions in v2.5 ? 

Thanks

**Comment by tidyui** [comment]
Not at the moment. Since Identity is really just an optional add-on module we haven’t invested that much time on it so far, but I’d really like to see an expansion on it. One way would be to add a second UserInformation entity which is template based, just like pages, posts & sites. Then you could select the user type on the user and get the fields available for that type.

This would give the possibility to have different sets of user info for users in your front application and the administrators.

Regards

**Comment by vjacquet** [comment]
Hello,

It is possible to implement the UI for users and roles using UserManager<TUser>, IUserStore<TUser>, RoleManager<TRole> and IRoleStore<TRole>, without the need of the concrete Piranha.AspNetCore.Identity.Data.User and Piranha.AspNetCore.Identity.Data.Role types.

it will not give you an interface to see and edit custom data but it will allow you to use your own User and Role classes.

I have a [proof of concept](https://github.com/PiranhaCMS/piranha.core/compare/master...vjacquet:features/identity), I can PR it if you are interested.

There are still some details to discuss, such as should the manager all the users or only users with a given claim or role. If the manager displays only a subset of the users, should it have a function to promote a user.

Regards

**Comment by vector-man** [comment]
@tidyui would the idea you propose get you custom user data editing in the backend?

**Comment by tidyui** [comment]
Yes, the idea with the proposed solution is that you would be able to different types of users, just like you can add different types of pages or posts with their own set of regions and fields. With this in place you could easily setup different user types and administrate them from the backend.

However, it seems most people use or create their own user management code, so there hasn't really been any focus on this idea.
